experimental/class-brand-check-expression.md

```diff
@@ -0,0 +1,28 @@
+# [class.hasInstance](https://github.com/tc39/proposal-class-brand-check)
+
+## ClassExpression
+
+```js
+interface Class <: Node {
+    type: "Class"
+}
+```
+A class pseudo-expression.
+
+```js
+extend interface CallExpression {
+    callee: Class // currently only "class"
+    arguments: [ Expression ]
+}
+```
```

Comment:
Some quick questions since the current proposal does not have a syntax spec.

Is
```js
class.hasInstance?.()
```
valid? If not we should add that `optional` must be `false`.

Is
```js
const hasInstance = class.hasInstance;
hasInstance(instance)
```
valid? If so we can't extend `CallExpression` because `class.hasInstance` itself can form an expression, just like `import.meta` and `function.sent` and thus not limited to be a `callee` of `CallExpression`.

Reply:
There is a gay who has submitted a [spec](https://github.com/tc39/proposal-class-brand-check/pull/3) about the proposal.
But it has not been merged.
I think I need to confirm with the mantiner of the proposal whether it needs to support ChainExpression.

Reply:
For reviewers: see https://github.com/tc39/proposal-class-brand-check/pull/3/files#diff-181371b08d71216599b0acccbaabd03c306da6de142ea6275c2135810999805aR173 for the proposed syntax.

```
CallExpression[Yield, Await] :
        CoverCallExpressionAndAsyncArrowHead[?Yield, ?Await] #callcover
        SuperCall[?Yield, ?Await]
        ImportCall[?Yield, ?Await]
        <ins>BrandCheckCall[?Yield, ?Await]</ins>
        CallExpression[?Yield, ?Await] Arguments[?Yield, ?Await]
        CallExpression[?Yield, ?Await] `[` Expression[+In, ?Yield, ?Await] `]`
        CallExpression[?Yield, ?Await] `.` IdentifierName
        CallExpression[?Yield, ?Await] TemplateLiteral[?Yield, ?Await, +Tagged]
        CallExpression[?Yield, ?Await] `.` PrivateIdentifier

      <ins>BrandCheckCall[Yield, Await] :</ins>
        `class` `.` `hasInstance` `(` AssignmentExpression[+In, ?Yield, ?Await] `)`
```

Reply:
The question is whether we want `class.hasInstance` be a real function/method. For example, `eval()` is a function but `import()` is not. As the author of the proposal, I prefer `import()` way because it's much simpler. If it was a method/function, we need to specify many behavior (such as whether it has receiver, aka. bound method or unbound), though we can specify the semantic anyway, it's hard to match everyone's expectation (a good/bad example is the trouble of `eval?.()` semantic --- indirect or direct eval). It seems make `class.hasInstance` be a real function/method do not have any real benefit, just introduce troubles.